treeherder/log_parser/intermittents.py

```diff
@@ -0,0 +1,134 @@
+from treeherder.model.models import Group, GroupStatus, Job, Push
+
+
+def check_and_mark_intermittent(job_id):
+    current_job = Job.objects.get(id=job_id)
+    jtname = current_job.job_type.name.strip("-cf")
+    ids = [current_job.push.id]
+
+    try:
+        _ = int(jtname.split("-")[-1])
+        jtname = "-".join(jtname.split("-")[:-1])
+    except ValueError:
+        pass
+
+    # if we are not on try, look at recent history
+    if current_job.repository.id != 4:
+        # get list of pushes
+        ids = Push.objects.filter(repository__id=current_job.repository.id).values("id")[:20]
+
+    all_groups = (
+        Group.objects.filter(
+            job_logs__job__push__id__in=ids,
```

Comment:
Could this be done with a push range and repository ID instead? A check against a set of integers looks not optimizable for the database (not sure if push ID not directly in the Group model is already an issue).

Reply:
yes, I think >X and <Y would be doable, there might be a smarter way with django like `__in_range=(x,y)` or some syntax.  let me dig into that

Reply:
yes `__range(x,y)`;  it is inclusive of x and y as well as everything in between.